deeppavlov/models/bert/bert_as_summarizer.py

```diff
@@ -0,0 +1,216 @@
+            from ru_sent_tokenize import ru_sent_tokenize
+            self.sent_tokenizer = ru_sent_tokenize
+        else:
+            from nltk import sent_tokenize
+            self.sent_tokenizer = sent_tokenize
+
+        self.sess_config = tf.ConfigProto(allow_soft_placement=True)
+        self.sess_config.gpu_options.allow_growth = True
+        self.sess = tf.Session(config=self.sess_config)
+
+        self._init_graph()
+
+        self.sess.run(tf.global_variables_initializer())
+
+        if pretrained_bert is not None:
+            pretrained_bert = str(expand_path(pretrained_bert))
+
+            if tf.train.checkpoint_exists(pretrained_bert):
+                logger.info('[initializing model with Bert from {}]'.format(pretrained_bert))
+                tvars = tf.trainable_variables()
+                assignment_map, _ = get_assignment_map_from_checkpoint(tvars, pretrained_bert)
+                tf.train.init_from_checkpoint(pretrained_bert, assignment_map)
+
+    def _init_graph(self):
+        self._init_placeholders()
+
+        self.bert = BertModel(config=self.bert_config,
+                              is_training=self.is_train_ph,
+                              input_ids=self.input_ids_ph,
+                              input_mask=self.input_masks_ph,
+                              token_type_ids=self.token_types_ph,
+                              use_one_hot_embeddings=False,
+                              )
+        # next sentence prediction head
+        with tf.variable_scope("cls/seq_relationship"):
+            output_weights = tf.get_variable(
+                "output_weights",
+                shape=[2, self.bert_config.hidden_size],
+                initializer=create_initializer(self.bert_config.initializer_range))
+            output_bias = tf.get_variable(
+                "output_bias", shape=[2], initializer=tf.zeros_initializer())
+
+        nsp_logits = tf.matmul(self.bert.get_pooled_output(), output_weights, transpose_b=True)
+        nsp_logits = tf.nn.bias_add(nsp_logits, output_bias)
+        self.nsp_probs = tf.nn.softmax(nsp_logits, axis=-1)
+
+    def _init_placeholders(self):
+        self.input_ids_ph = tf.placeholder(shape=(None, None), dtype=tf.int32, name='ids_ph')
+        self.input_masks_ph = tf.placeholder(shape=(None, None), dtype=tf.int32, name='masks_ph')
+        self.token_types_ph = tf.placeholder(shape=(None, None), dtype=tf.int32, name='token_types_ph')
+
+        self.is_train_ph = tf.placeholder_with_default(False, shape=[], name='is_train_ph')
+
+    def _build_feed_dict(self, input_ids, input_masks, token_types):
+        feed_dict = {
+            self.input_ids_ph: input_ids,
+            self.input_masks_ph: input_masks,
+            self.token_types_ph: token_types,
+        }
+        return feed_dict
+
+    def _get_nsp_predictions(self, sentences: List[str], candidates: List[str]):
+        """Compute NextSentence probability for every (sentence_i, candidate_i) pair.
```

Comment:
Is there intentionally the same index i or you mean:
```Compute NextSentence probability for every (sentence_i, candidate_j) pair.```
?

Reply:
Yes, `(sentence_i, candidate_i)`, you can think of `sentences` and `candidates` as batches.

Reply:
Do you mean that it is better to remove `every` from docstring?